### PR TITLE
NAS-125045 / 24.04 / Add new builtin groups for builtin roles

### PR DESCRIPTION
**Pull request by themylogin**

### CODE REVIEW

**Comment by themylogin** [comment]
https://ci.tn.ixsystems.net/jenkins/job/TrueNAS%20SCALE%20-%20Unstable/job/API%20Tests%20-%20TrueNAS%20SCALE%20with%20middleware%20reinstalled%20from%20a%20branch%20Incremental/1618/

**Comment by themylogin** [comment]
time 4:00

File: src/middlewared/middlewared/alembic/versions/24.04/2024-01-15_20-10_builtin_roles_groups.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    conn.execute("INSERT INTO account_bsdgroups (bsdgrp_gid, bsdgrp_group, bsdgrp_builtin, bsdgrp_smb,"
+                 "bsdgrp_sudo_commands, bsdgrp_sudo_commands_nopasswd) VALUES (951, 'truenas_readonly_administrators', "
+                 "1, 1, '[]', '[]')")
+
+    conn.execute("INSERT INTO account_bsdgroups (bsdgrp_gid, bsdgrp_group, bsdgrp_builtin, bsdgrp_smb,"
+                 "bsdgrp_sudo_commands, bsdgrp_sudo_commands_nopasswd) VALUES (952, 'truenas_sharing_administrators', "
+                 "1, 1, '[]', '[]')")
+
+    op.execute(
+        "UPDATE account_privilege SET builtin_name = 'READONLY_ADMININSTATOR', name = 'Read-Only Administator', "
```
Comment (yocalebo) [inline src/middlewared/middlewared/alembic/versions/24.04/2024-01-15_20-10_builtin_roles_groups.py:32]:
```suggestion
        "UPDATE account_privilege SET builtin_name = 'READONLY_ADMININSTRATOR', name = 'Read-Only Administator', "
```

File: src/middlewared/middlewared/alembic/versions/24.04/2024-01-15_20-10_builtin_roles_groups.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        "local_groups = '[951]', roles = '[\"READONLY_ADMIN\"]' "
+        "WHERE builtin_name = 'READONLY'"
+    )
+    op.execute(
+        "UPDATE account_privilege SET builtin_name = 'SHARING_ADMINISTRATOR', name = 'Sharing Administator', "
```
Comment (yocalebo) [inline src/middlewared/middlewared/alembic/versions/24.04/2024-01-15_20-10_builtin_roles_groups.py:37]:
```suggestion
        "UPDATE account_privilege SET builtin_name = 'SHARING_ADMINISTRATOR', name = 'Sharing Administrator', "
```

File: src/middlewared/middlewared/plugins/account_/privilege.py [hunk 3]
```diff
@@ -15,8 +15,8 @@
 
 class BuiltinPrivileges(enum.Enum):
     LOCAL_ADMINISTRATOR = "LOCAL_ADMINISTRATOR"
-    READONLY = "READONLY"
-    SHARING_MANAGER = "SHARING_MANAGER"
+    READONLY_ADMININSTATOR = "READONLY_ADMININSTATOR"
```
Comment (yocalebo) [inline src/middlewared/middlewared/plugins/account_/privilege.py:18]:
```suggestion
    READONLY_ADMINISTRATOR = "READONLY_ADMINISTRATOR"
```

File: src/middlewared/middlewared/plugins/account_/privilege.py [hunk 4]
```diff
@@ -143,7 +143,7 @@ async def do_update(self, id_, data):
                         "None of the members of these groups has password login enabled. At least one grantee of "
                         "the \"Local Administrator\" privilege must have password login enabled."
                     )
-            elif builtin_privilege == BuiltinPrivileges.READONLY:
+            elif builtin_privilege == BuiltinPrivileges.READONLY_ADMININSTATOR:
```
Comment (yocalebo) [inline src/middlewared/middlewared/plugins/account_/privilege.py:146]:
```suggestion
            elif builtin_privilege == BuiltinPrivileges.READONLY_ADMINISTRATOR:
```

**Review by yocalebo — CHANGES_REQUESTED** [review]

**Comment by yocalebo** [comment]
There will be conflicts after this PR is merged so holding on on approving right now: https://github.com/truenas/middleware/pull/12893

**Review by anodos325 — APPROVED** [review]

**Review by yocalebo — APPROVED** [review]

**Pull request merged**